packages/editor/src/components/template-validation-notice/index.js

```diff
@@ -1,63 +1,62 @@
 /**
  * WordPress dependencies
  */
-import { Notice } from '@wordpress/components';
+import {
+	Notice,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 
-function TemplateValidationNotice( { isValid, ...props } ) {
+export default function TemplateValidationNotice() {
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+	const { isValid } = useSelect( ( select ) => {
+		return {
+			isValid: select( blockEditorStore ).isValidTemplate(),
+		};
+	}, [] );
```

Comment:
```suggestion
	const isValid = useSelect( ( select ) => {
		return select( blockEditorStore ).isValidTemplate();
	}, [] );
```

Perhaps this is simpler?

Reply:
I don't have a strong preference here as long as it's correct :D